src/storage/v2/disk/storage.cpp

```diff
@@ -0,0 +1,1703 @@
+  other.commit_timestamp_.reset();
+}
+
+DiskStorage::DiskAccessor::~DiskAccessor() {
+  if (is_transaction_active_) {
+    Abort();
+  }
+
+  FinalizeTransaction();
+}
+
+/// NOTE: This will create Delta object which will cause deletion of old key entry on the disk
+std::optional<storage::VertexAccessor> DiskStorage::DiskAccessor::LoadVertexToMainMemoryCache(
+    const rocksdb::Slice &key, const rocksdb::Slice &value) {
+  auto main_storage_accessor = vertices_.access();
+
+  const std::string key_str = key.ToString();
+  storage::Gid gid = Gid::FromUint(std::stoull(utils::ExtractGidFromKey(key_str)));
+  if (VertexExistsInCache(main_storage_accessor, gid)) {
+    return std::nullopt;
+  }
+  std::vector<LabelId> labels_id = utils::DeserializeLabelsFromMainDiskStorage(key_str);
+  return CreateVertex(main_storage_accessor, gid, labels_id,
+                      utils::DeserializePropertiesFromMainDiskStorage(value.ToStringView()),
+                      CreateDeleteDeserializedObjectDelta(&transaction_, key.ToString()));
+}
+
+std::optional<storage::VertexAccessor> DiskStorage::DiskAccessor::LoadVertexToLabelIndexCache(
+    const rocksdb::Slice &key, const rocksdb::Slice &value, Delta *index_delta,
+    utils::SkipList<storage::Vertex>::Accessor index_accessor) {
+  storage::Gid gid = Gid::FromUint(std::stoull(utils::ExtractGidFromLabelIndexStorage(key.ToString())));
+  if (VertexExistsInCache(index_accessor, gid)) {
+    return std::nullopt;
+  }
+
+  const std::string value_str{value.ToString()};
+  const auto labels{utils::DeserializeLabelsFromLabelIndexStorage(value_str)};
+  return CreateVertex(index_accessor, gid, labels, utils::DeserializePropertiesFromLabelIndexStorage(value_str),
+                      index_delta);
+}
+
+std::optional<storage::VertexAccessor> DiskStorage::DiskAccessor::LoadVertexToLabelPropertyIndexCache(
+    const rocksdb::Slice &key, const rocksdb::Slice &value, Delta *index_delta,
+    utils::SkipList<storage::Vertex>::Accessor index_accessor) {
+  storage::Gid gid = Gid::FromUint(std::stoull(utils::ExtractGidFromLabelPropertyIndexStorage(key.ToString())));
+  if (VertexExistsInCache(index_accessor, gid)) {
+    return std::nullopt;
+  }
+
+  const std::string value_str{value.ToString()};
+  const auto labels{utils::DeserializeLabelsFromLabelPropertyIndexStorage(value_str)};
+  return CreateVertex(index_accessor, gid, labels,
+                      utils::DeserializePropertiesFromLabelPropertyIndexStorage(value.ToString()), index_delta);
+}
+
+std::optional<EdgeAccessor> DiskStorage::DiskAccessor::DeserializeEdge(const rocksdb::Slice &key,
+                                                                       const rocksdb::Slice &value) {
+  const auto edge_parts = utils::Split(key.ToStringView(), "|");
+  const Gid edge_gid = Gid::FromUint(std::stoull(edge_parts[4]));
+
+  auto edge_acc = edges_.access();
+  auto res = edge_acc.find(edge_gid);
+  if (res != edge_acc.end()) {
+    return std::nullopt;
+  }
```

Comment:
Can we do that check before somewhere, so this function only does deserialization of edge from rocksdb